src/integrationtests/java/com/aws/greengrass/integrationtests/ipc/ShadowIPCTest.java

```diff
@@ -200,19 +191,55 @@ static void stopKernel() {
         kernel.shutdown();
     }
 
+    @Test
+    @Order(0)
+    void GIVEN_shadow_manager_running_WHEN_I_stress_shadow_manager_THEN_no_errors_occur() throws Exception {
+        try (EventStreamRPCConnection connection = IPCTestUtils.getEventStreamRpcConnection(kernel, "DoAll");
+             GreengrassCoreIPCClientV2 ipcClient = GreengrassCoreIPCClientV2.builder()
+                     .withClient(new GreengrassCoreIPCClient(connection)).withoutExecutor().build()) {
+            String updateDocumentStr =
+                    "{\"state\":{\"reported\":{\"color\":{\"r\":255,\"g\":255,\"b\":255}," + "\"SomeKey\":\"time\"}}}";
+
+            // Stress test for updating shadow and syncing it to cloud.
+            for (int i = 0; i < 1000; i++) {
+                final long currentEpochSeconds = Instant.now().toEpochMilli();
+
+                UpdateThingShadowRequest updateThingShadowRequest = new UpdateThingShadowRequest();
+                updateThingShadowRequest.setThingName(MOCK_THING_NAME);
+                updateThingShadowRequest.setShadowName(SHADOW_NAME_1);
+                updateThingShadowRequest.setPayload(
+                        updateDocumentStr.replaceAll("time", String.valueOf(currentEpochSeconds)).getBytes(UTF_8));
+
+                CompletableFuture<UpdateThingShadowResponse> fut =
+                        ipcClient.updateThingShadowAsync(updateThingShadowRequest);
+                fut.get(90, TimeUnit.SECONDS);
+            }
+
+            // Wait for any pending operations to complete before shutting down.
+            Thread.sleep(5000);
+
+            ipcClient.deleteThingShadow(new DeleteThingShadowRequest()
```

Comment:
Just for my understanding what is this line testing?

Reply:
Not testing, this is just resetting the state back to beginning. Tests after this one have expectations on what the shadow version will be (1, not 1,000).